src/components/pages/Offers/Offer/Stocks/StockItem/StockItem.jsx

```diff
@@ -44,20 +44,20 @@ const StockItem = ({
   const [price, setPrice] = useState(initialStock.price)
   const [totalQuantity, setTotalQuantity] = useState(initialStock.quantity)
   const [isActivationCodesDialogOpen, setIsActivationCodesDialogOpen] = useState(false)
-  const [activationCodes, setActivationCodes] = useState(initialStock.activationCodes)
+  const [activationCodes, setActivationCodes] = useState(initialStock.activationCodes || [])
   const [activationCodesExpirationDatetime, setActivationCodesExpirationDatetime] = useState(
     initialStock.activationCodesExpirationDatetime
   )
 
   const hasActivationCodes = isNewStock
     ? activationCodes.length > 0
-    : initialStock.hasActivationCodes
+    : !!initialStock.hasActivationCodes
```

Comment:
```suggestion
    : Boolean(initialStock.hasActivationCodes)
```

C'est plus explicite je trouve

Reply:
yes

Reply:
Corrigé